src/ip/Address.cc

```diff
@@ -644,7 +644,7 @@ Ip::Address::getAddrInfo(struct addrinfo *&dst, int force) const
         dst->ai_protocol = IPPROTO_UDP;
 
     if (force == AF_INET6 || (force == AF_UNSPEC && isIPv6()) ) {
-        dst->ai_addr = (struct sockaddr*)new sockaddr_in6;
+        dst->ai_addr = (struct sockaddr*)new sockaddr_storage;
```

Comment:
Please replace this with a call to `InitAddr` to minimize the places doing new/delete.
```suggestion
        InitAddr(dst);
```